pkg/runner/healthcheck_common.go

```diff
@@ -0,0 +1,79 @@
+package runner
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ipfs/testground/pkg/api"
+	"golang.org/x/sync/errgroup"
+)
+
+type Checker func() bool
+type Fixer func() error
+
+type Healthchecked interface {
+	Enlist(name string, c Checker, f Fixer)
+	RunChecks(ctx context.Context, fix bool)
+}
+
+type toDoElement struct {
+	Name    string
+	Checker Checker
+	Fixer   Fixer
+}
+
+type ErrgroupHealthcheckHelper struct {
```

Comment:
I thought there might be some conditions under which concurrent fixes might be inappropriate so there might be multiple Healthchecked interface implementations. In this one, the implementation of RunChecks closely resembles the change to local_exec runner. 
See https://github.com/ipfs/testground/pull/703 and  https://github.com/ipfs/testground/pull/733
My intention is to move that logic here, easily convert the local exec runner to use this.

I suspect the docker runner can use this helper also